app/routes/task_routes.py

```diff
@@ -1 +1,118 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response, Response
+from .route_utilities import validate_model
+from app.models.task import Task
+from datetime import datetime
+from ..db import db
+import requests
+import os
+
+
+SLACK_API_URL = os.environ["SLACK_API_URL"]
+SLACK_BOT_TOKEN = os.environ["SLACK_BOT_TOKEN"]
+
+bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    sort_method = request.args.get('sort')
+
+    if sort_method and sort_method == "asc":
+        query = query.order_by(Task.title.asc())
+    if sort_method and sort_method == "desc":
+        query = query.order_by(Task.title.desc())
+
+    tasks = db.session.scalars(query)
+    tasks_response = [task.to_dict() for task in tasks]
+    
+    return tasks_response
+
+
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task.from_dict(request_body)
+
+    except KeyError as error:
+        response = {"details": f"Invalid data"}
+        abort(make_response(response, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+
+@bp.get("/<task_id>")
+def get_one_task(task_id):
+    task = validate_model(Task, task_id)
+    return {"task": task.to_dict()}
+
+
+@bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.add(task)
+    db.session.commit()
+
+    return Response(status=204, mimetype="application/json")
```

Comment:
In `task_routes.py` I updated lines 65, 74, 87, and 100. The rest of the implementation is unchanged.